scipy/io/arff/tests/test_arffread.py

```diff
@@ -195,6 +198,18 @@ def test_year_attribute(self):
         ], dtype='datetime64[Y]')
 
         assert_array_equal(self.data["attr_year"], expected)
+    
+    def test_yeari_attribute(self):
+        expected = np.array([
+            '99-01-31',
+            '04-12-01',
+            '17-04-28',
+            '00-09-10',
+            '13-11-30',
+            '31-10-15'
+        ], dtype='datetime64[Y]')
```

Comment:
Are "years" the smallest units you're expecting to get back here? If I switch to days it is a little closer:

```
actual: ['1999-01-31' '2004-12-01' '2017-04-28' '2000-09-10' '2013-11-30'
 '2031-10-15']
expected: ['0099-01-31' '0004-12-01' '0017-04-28' '0000-09-10' '0013-11-30'
 '0031-10-15']
datetime64[D] datetime64[D]
```

Looks like `31` is getting converted to `2031`, but I suppose we're expecting only the two-digit year to be extracted unambiguously here.

Reply:
This seems to be what happens under the hood:

```python
>>> datetime.datetime.strptime('31-10-15', '%y-%m-%d')
datetime.datetime(2031, 10, 15, 0, 0)
```

Reply:
I could be wrong, but the prospects for returning a `datetime` object with the century somehow stripped out are maybe not great cc @eric-wieser ?

The question of what we expect to get back here in SciPy is actually not immediately clear to me, but perhaps it would actually match the century restoration behavior of the std lib?